network/alsp/manager/manager.go

```diff
@@ -337,12 +337,28 @@ func (m *MisbehaviorReportManager) onHeartbeat() error {
 			// each time we decay the penalty by the decay speed, the penalty is a negative number, and the decay speed
 			// is a positive number. So the penalty is getting closer to zero.
 			// We use math.Min() to make sure the penalty is never positive.
-			record.Penalty = math.Min(record.Penalty+record.Decay, 0)
+			m.logger.Trace().
+				Hex("identifier", logging.ID(id)).
+				Uint64("cutoff_counter", record.CutoffCounter).
+				Bool("disallow_listed", record.DisallowListed).
+				Float64("penalty", record.Penalty).
+				Msg("heartbeat interval, pulled the spam record for decaying")
+			record.Penalty = m.decayFunc(record)
+			m.logger.Trace().
+				Hex("identifier", logging.ID(id)).
+				Uint64("cutoff_counter", record.CutoffCounter).
+				Bool("disallow_listed", record.DisallowListed).
+				Float64("penalty", record.Penalty).
+				Msg("heartbeat interval, spam record penalty adjusted by decay function")
 
 			// TODO: this can be done in batch but at this stage let's send individual notifications.
 			//       (it requires enabling the batch mode end-to-end including the cache in middleware).
 			if record.Penalty == float64(0) && record.DisallowListed {
 				record.DisallowListed = false
+
+				//// after fully decaying the penalty, update decay for next disallow listing
+				//record.UpdateDecay()
+
```

Comment:
Please remove commented code if it is not needed.

Reply:
7bda8d2e70703a546189be1b404e2a10da6cf79f